packages/remote-view/src/utils/getUrls.ts

```diff
@@ -0,0 +1,40 @@
+// Strip trailing slashes
+function normalizeBaseUrl(baseUrl: string) {
+  if (baseUrl.endsWith('/')) {
+    return baseUrl.slice(0, baseUrl.length - 1);
+  }
+
+  return baseUrl;
+}
+
+// Safe path to package.json
+export function getRemotePackageJsonUrl(baseUrl: string) {
+  const normalizedBaseUrl = normalizeBaseUrl(baseUrl);
+  return `${normalizedBaseUrl}/package.json`;
+}
+
+// Safe path to `module` or `style` defined in a package.json
+export function getRemoteAssetUrl(
```

Comment:
This doesn't seem to support paths as I'd hoped. Example: supply `http://localhost:8484/esm-view-card` as the `normalizedBaseUrl` and `/package.json` as the `assetPath` = `http://localhost:8484/package.json` (incorrect)

Reply:
That's expected isn't it? When the `assetPath` is prefixed with a `/`, it should be at the root of the host of `normalizedBaseUrl`.

```
assetPath       normalizedBaseUrl     result
/path               http://host/a/b/c          http://host/path
./path              http://host/a                http://host/path
./path              http://host/a/               http://host/a/path
../path             http://host/a/b/c          http://host/a/path
../path             http://host/a/b/c/         http://host/a/b/path
```

Reply:
That's expected for the URL constructor but it doesn't solve the problem we are trying to solve here. This simplification of the get-url utils would require splitting origins from paths, which come from user input and aren't currently validated.

At the moment users are expected to provide a path to the base URL of an ESM View (e.g. `https://cdn.example.com/my-esm-card-view`.) To easily get `package.json` from that input and the URL constructor, I'd need to split `https://cdn.example.com/` from `/my-esm-card-view` and call:

```javascript
const userInput = "https://cdn.example.com/my-esm-card-view";

// Incorrect result
new URL("/package.json", userInput);

// Correct result
new URL("/my-esm-card-view/package.json", "https://cdn.example.com")
```
This isn't so hard to solve itself but we don't prevent user input such as:

- `cdn.example.com/my-esm-card-view`
- `/my-esm-card-view`
- `localhost:8080/my-esm-card-view`
- `http://localhost:8080/my-esm-card-view`
- `https://cdn.example.com/my-esm-card-view`

That said, if we assume users follow the patterns used in the docs, we could throw if a non-absolute-including-protocol URL is supplied

Reply:
Please could you create a table of inputs and expected outputs?
Perhaps the system that is generating a path of `/package.json` be changed to generate `./package.json` if it intends to serve a relative path rather than a root path.